### DISCUSSION TITLE
How to add more than one relation on a single model? (MongoDB)

### DISCUSSION THREAD

**Discussion by mithatercan** [prisma/prisma#14574] [Q&A] [answered]
Hello,
I'm attempting to create a relation using MongoDB's prisma model. However, I am unable to assign a value to the parameter. And I've included my code below to demonstrate how I try to construct nested models and get this problem. How is it possible to do like this?

Thanks in advance.

> Unknown arg type in data.type for type CardCreateInput.

```javascript
prisma.guest.create({
    data: {
      ...rest,
      card: {
        create: {
          no: cardNo,
          type: {
            connect: {
              id: cardTypeId,
            },
          },
        },
      },
      client: {
        connect: {
          username: username,
        },
      },
    },
  })
```

```
model Guest {
  id          String   @id @default(auto()) @map("_id") @db.ObjectId
  name        String
  address     String
  phoneNumber String
  gender      String
  birthday    String
  isActive    Boolean? @default(true)
  actions     Action[] @relation("Actions")
  cards       Card[]   @relation("Cards")
  client      Client   @relation("Guests", fields: [clientId], references: [id])
  clientId    String   @db.ObjectId
  createdAt   DateTime @default(now())
  updatedAt   DateTime @default(now())
}

model CardModel {
  id        String   @id @default(auto()) @map("_id") @db.ObjectId
  name      String
  value     Int
  type      CardType
  client    Client   @relation("CardModels", fields: [clientId], references: [id])
  clientId  String   @db.ObjectId
  cards     Card[]   @relation("CardModel")
  createdAt DateTime @default(now())
  updatedAt DateTime @default(now())
}

model Card {
  id          String    @id @default(auto()) @map("_id") @db.ObjectId
  no          String    @unique
  model        CardModel @relation("CardModel", fields: [typeId], references: [id])
  modelId      String    @db.ObjectId 
  guest       Guest     @relation("Cards", fields: [guestId], references: [id])
  guestId     String    @unique @db.ObjectId
}
```

**Comment by janpio** [comment, marked as answer]
Your model `Card` has no field `type`. It seems to be in `CardModel`, so you would need to go from Card to CardModel first to then be able to fill the relation to CardType.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
